tests/udt/src/lib.rs

```diff
@@ -36,7 +36,127 @@ impl Contract {
+        let e = Env::default();
+        let udt = UdtStruct {
+            a: 10,
+            b: 12,
+            c: vec![&e, 1],
+        };
+        let bin = udt.serialize(&e);
+        assert_eq!(bin, {
+            let mut bin = Binary::new(&e);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(4);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(1);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(1);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(3);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(5);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(1);
+            bin.push(97);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(10);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(5);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(1);
+            bin.push(98);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(12);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(5);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(1);
+            bin.push(99);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(4);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(1);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(1);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(0);
+            bin.push(1);
+            bin
+        })
+    }
```

Comment:
If you prefer not writing these repetitive lines, the `binary_new_from_linear_memory` may serve you. Here is the example: 
https://github.com/stellar/rs-stellar-contract-sdk/pull/133/files#diff-4a1250d4e98a8c8930b456dfeb1b323756724a164cba250df28c0d71bbabe722R11-R12

Reply:
Thanks, I'll follow up with fixing that up.